File: src/content/3.11/code/scala/snippet10.scala
implicit def freeFunctor[F[_]] = new Functor[FreeF[F, ?]] {
  def fmap[A, B](g: A => B)(fa: FreeF[F, A]): FreeF[F, B] = {
    new FreeF[F, B] {
      def h[I]: I => B = g compose fa.h
      def fi[I]: F[I] = fi
    }
  }
}

File: src/content/3.11/code/ocaml/snippet10.ml
module FreeFunctor (F : sig
  type 'a f
end) : Functor = struct
  module type F = FreeF with type 'a f = 'a F.f

  type 'a t = (module F with type a = 'a)

  let fmap
      (type a' b')
      (f : a' -> b')
      (module FF : F with type a = a')
    =
    (module struct
      type 'a f = 'a F.f
      type a = b'
      type i = FF.i

      let h i = f (FF.h i)
      let fi = FF.fi
    end : F
      with type a = b')
  ;;
end
